src/algebra/field/mod.rs

```diff
@@ -74,3 +74,11 @@ pub trait FiniteField: Finite + Field {
     Self::PRIMITIVE_ELEMENT.pow(pow)
   }
 }
+
+/// traits that define some functions usable in signatures
+pub trait FieldExt: Sized {
```

Comment:
Still not a fan of adding an additional trait. But we can always refactor later.